### PR TITLE
[cherry-pick] : cherry-pick 17161 17020 16522 to save memory useage for dup check and clean up code

### PR DESCRIPTION
**Pull request by jensenojs**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #https://github.com/matrixorigin/matrixone/issues/16375

## What this PR does / why we need it:
optmize duplicate check memory usage for sql like insert into t1 selct from t2

### CODE REVIEW

**Comment by jensenojs** [comment]
提这个cherry-pick的原因是1.2-dev上也有在`insert into select`大数据量场景下因为gc导致耗时多了一倍的问题, 因此先要想办法把这里的内存占用给降下来, 问题能否修复需要再验证, 可能还需要再cp几个其他人的pr.

https://github.com/matrixorigin/matrixone/issues/17143#issuecomment-2190431895

***

等相关的内存占用的优化, 和统计信息的确认能过之后, 再请大家review这个pr, 现在弄中移的case

**Review by m-schen — APPROVED** [review]

**Comment by jensenojs** [comment]
```sql
create table t0 (a int, b int, c int, primary key (a, b));
create table t1 like t0;
create table t2 like t0;
insert into t1 select result, result, result from generate_series(1,200000000) g;
```

在这个pr上的测试结果 : 耗时 : 2 min 58.11

heap : 

TODO

在main上的测试结果 :  耗时 : 4 min 1.55

heap : 

TODO

**Review by badboynt1 — APPROVED** [review]

File: proto/plan.proto [hunk 1]
```diff
@@ -834,12 +834,14 @@ message Node {
...
 	repeated Expr onUpdateExprs = 53;
 	OriginTableMessageForFuzzy fuzzymessage = 54;
+	bool ifInsertFromUnique = 55;
 
 	//for message
-	repeated msgHeader send_msg_list = 55;
-	repeated msgHeader recv_msg_list = 56;
+	repeated msgHeader send_msg_list = 56;
+	repeated msgHeader recv_msg_list = 57;
+
```
Comment (zhangxu19830126) [inline proto/plan.proto:842]: use [(gogoproto.nullable) = false] to avoid pointer object

**Review by zhangxu19830126 — CHANGES_REQUESTED** [review]

File: proto/plan.proto (hunk 1 above)
Comment (jensenojs) [reply proto/plan.proto:842]: fix

**Review by ouyuanning — APPROVED** [review]

**Review by zhangxu19830126 — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]

**Review by sukki37 — APPROVED** [review]

**Pull request merged**